src/runtime/TemporalObject.cpp

```diff
@@ -202,17 +203,98 @@ std::map<std::string, std::string> TemporalObject::parseValidIso8601String(Execu
         nanoSecond = fraction.substr(6, 3);
     }
 
-    if (!TemporalPlainDate::isValidISODate(state, Value(year.c_str()), Value(month.c_str()), Value(day.c_str()))) {
+    if (!TemporalPlainDate::isValidISODate(state, Value(year.c_str()).asInt32(), Value(month.c_str()).asInt32(), Value(day.c_str()).asInt32())) {
         ErrorObject::throwBuiltinError(state, ErrorObject::RangeError, "Invalid ISO Date");
     }
 
-    if (!TemporalPlainTime::isValidTime(state, Value(hour.c_str()), Value(minute.c_str()), Value(second.c_str()), Value(millisecond.c_str()), Value(microSecond.c_str()), Value(nanoSecond.c_str()))) {
+    if (!TemporalPlainTime::isValidTime(state, Value(hour.c_str()).asInt32(), Value(minute.c_str()).asInt32(), Value(second.c_str()).asInt32(), Value(millisecond.c_str()).asInt32(), Value(microSecond.c_str()).asInt32(), Value(nanoSecond.c_str()).asInt32())) {
         ErrorObject::throwBuiltinError(state, ErrorObject::RangeError, "Invalid time");
     }
 
     return std::map<std::string, std::string>{ { "Year", year }, { "Month", month }, { "Day", day }, { "Hour", hour }, { "Minute", minute }, { "Second", second }, { "Millisecond", millisecond }, { "Microsecond", microSecond }, { "Nanosecond", nanoSecond }, { "Calendar", calendar } };
 }
 
+std::map<std::string, std::string> TemporalObject::parseTemporalDateString(ExecutionState& state, std::string isoString)
+{
+    bool good = false;
+
+    size_t pos = isoString.find('[');
+    std::string isoDate = isoString.substr(0, pos);
+    std::string extra;
+
+    if (pos != std::string::npos) {
+        extra = isoString.substr(pos);
+    }
+
+    pos = isoDate.find('T');
+
+    if (pos != std::string::npos) {
+        std::replace(isoDate.begin(), isoDate.end(), 'T', ' ');
+    }
+
+    pos = isoDate.find(' ');
+    std::string date = isoString.substr(0, pos);
+    std::string time = isoString.substr(pos + 1);
+    std::string year;
+    std::string month;
+    std::string week;
+    std::string day;
+    pos = date.find('-');
+
+    if (pos != std::string::npos) {
+        if (pos == 0 && date.length() > 4) {
+            if (date.substr(pos + 1).find('-') == 0) {
+                date.erase(std::remove(date.begin(), date.end(), '-'), date.end());
+                if (date.length() == 4) {
+                    good = true;
```

Comment:
What about using a `goto`? This way we continue the checks even if there are fails.